### ISSUE TITLE
Relax linter "quotes" requirement 

### ISSUE THREAD

**Issue by migmartri** [kubernetes/helm#1037]
Currently the linter is too strict in terms of quoting, we should revisit in which cases this is a valid warning and in which other cases this is an inconvenience or even a problem.

```
[WARNING] templates/odoo-svc.yaml: wrap substitution functions in quotes or use the sprig "quote" function: : {{ .Release.Name }} -> : {{ .Release.Name  | quote }}
```

https://github.com/kubernetes/helm/issues/689#issuecomment-225097389

**Comment by sameersbn** [comment]
I was testing the use of using unquoted values for labels and selectors fields and noticed the same issues that we originally mentioned in https://github.com/kubernetes/helm/issues/689#issuecomment-225097389. 

For example the following will cause a deployment to fail with the msg `Error: release sweet-dachshund failed: unable to decode "": [pos 375]: json: expect char '"' but got char '2'`

```
  labels:
    build: 201608
```

The linter rules can be enforced using the `--strict` arg. At the moment the linter only warns the user about these with issues without causing the linter to fail. I think its good that chart developers are warned about this to avoid user issues.

**Comment by waisbrot** [comment]
Here's an example where linter should not complain about quotes:

```
apiVersion: "extensions/v1beta1"
kind: Deployment
metadata:
  name: my-deployment
  labels:
    heritage: {{ .Release.Service | quote }}
    release: {{ .Release.Name | quote }}
    chart: "{{.Chart.Name}}-{{.Chart.Version}}"
spec:
  replicas: {{default 1 .Values.replicas}}
...
```

kubectl will check the type of `replicas` and it _must_ be an integer. But the linter sees that I've got an unquoted integer and warns. The above code gets a linter warning but works. Changing the line to 

```
replicas: {{default 1 .Values.replicas | quote}}
```

satisfies the linter but kubectl will fail.

(I believe this is what @migmartri is also describing)

**Comment by lachie83** [comment]
More fodder. ContainerPorts are integers. Taken from this PR -- https://github.com/kubernetes/charts/pull/31

```
lachlanevenson@faux$ helm lint .
2016-08-17 11:26:48.701414 I | Preparing template spark/templates/spark-master-deployment.yaml
2016-08-17 11:26:48.702288 I | Preparing template spark/templates/spark-master-svc.yaml
2016-08-17 11:26:48.702582 I | Preparing template spark/templates/spark-webui-svc.yaml
2016-08-17 11:26:48.702839 I | Preparing template spark/templates/spark-worker-deployment.yaml
2016-08-17 11:26:48.703267 I | Preparing template spark/templates/spark-worker-hpa.yaml
2016-08-17 11:26:48.703539 I | Preparing template spark/templates/spark-zeppelin-deployment.yaml
2016-08-17 11:26:48.703912 I | Preparing template spark/templates/spark-zeppelin-svc.yaml
==> Linting .
[WARNING] templates/spark-master-deployment.yaml: wrap substitution functions in quotes or use the sprig "quote" function: : {{.Values.Master.ContainerPort}} -> : {{.Values.Master.ContainerPort | quote }}
[WARNING] templates/spark-master-svc.yaml: wrap substitution functions in quotes or use the sprig "quote" function: : {{.Values.Master.ServicePort}} -> : {{.Values.Master.ServicePort | quote }}
[WARNING] templates/spark-webui-svc.yaml: wrap substitution functions in quotes or use the sprig "quote" function: : {{.Values.WebUi.ServicePort}} -> : {{.Values.WebUi.ServicePort | quote }}
[WARNING] templates/spark-worker-deployment.yaml: wrap substitution functions in quotes or use the sprig "quote" function: : {{.Values.Worker.ContainerPort}} -> : {{.Values.Worker.ContainerPort | quote }}
[WARNING] templates/spark-zeppelin-deployment.yaml: wrap substitution functions in quotes or use the sprig "quote" function: : {{.Values.Zeppelin.ContainerPort}} -> : {{.Values.Zeppelin.ContainerPort | quote }}
[WARNING] templates/spark-zeppelin-svc.yaml: wrap substitution functions in quotes or use the sprig "quote" function: : {{.Values.Zeppelin.ServicePort}} -> : {{.Values.Zeppelin.ServicePort | quote }}
```

**Comment by migmartri** [comment]
Thank you everybody for the comments. Definitely the replicas and ContainerPorts examples are serious flaws of this rule. 

I think that as first step, we should disable this linter rule since:

* AFAIK, not quoting in most of the required cases, makes the chart instantiation to crash and the chart developer/user will notice. I'd be more worried if not having quotes will add unexpected behavior but I have not noticed any yet.
* We can start building this rule from the bottom up.

What I want to avoid is to follow a white-list approach based on "complex" YAML un-marshaling (ContainerPorts: of example should be integer) rules, without being sure that this adds more value than inconveniences.

What do you think?

**Comment by technosophos** [comment]
I believe this is now closed via #1077.
